### PR TITLE
test: [M3-8445] Add new test to confirm changes to the Object details drawer for OBJ Gen 2

### PR DESCRIPTION
**Pull request by AzureLatte**
## Description 📝
Add a Cypress integration test using mock API data to confirm changes to the Object details drawer for OBJ Gen 2.

## Changes  🔄
List any change relevant to the reviewer.
- Add new test in packages/manager/cypress/e2e/core/objectStorageGen2/bucket-object-gen2.spec.ts
- Add new util functions in packages/manager/cypress/support/intercepts/object-storage.ts

## How to test 🧪
```
yarn cy:run -s "cypress/e2e/core/objectStorageGen2/bucket-object-gen2.spec.ts"
```

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by dwiley-akamai — APPROVED** [review]
`bucket-object-gen2.spec.ts` passes ✅
Code review ✅

Approved pending resolution of the conflict in `packages/manager/cypress/support/intercepts/object-storage.ts`

File: packages/manager/cypress/e2e/core/objectStorageGen2/bucket-object-gen2.spec.ts [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+          .click();
+      }
+      // Close the Details drawer
+      cy.get('[data-qa-close-drawer="true"]').should('be.visible').click();
+    });
+  };
+
+  /**
+  
+     */
```
Comment (abailly-akamai) [inline packages/manager/cypress/e2e/core/objectStorageGen2/bucket-object-gen2.spec.ts:135]: Did you intend to have a comment here?

**Review by abailly-akamai — APPROVED** [review]
Looks good - tests passing on my end and code looks good ✅

**Comment by abailly-akamai** [comment]
@AzureLatte you have a conflict you'll want to resolve 👍

**Comment by AzureLatte** [comment]
Test failure  delete-placement-groups.spec.ts is not related with this change.

**Pull request merged**